core/controllers/admin.py

```diff
@@ -2261,6 +2262,29 @@ def put(self) -> None:
         self.render_json({})
 
 
+class RegenerateTopicSummariesHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handler to regenerate the summaries of all topics."""
+
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'PUT': {}
+    }
+
+    @acl_decorators.can_access_admin_page
+    def put(self) -> None:
```

Comment:
## Results
### 100 Topic Summary Models

| Attempt | Server response wait time |
| ---- | ---- |
| 1 | 5.88 s |
| 2 | 0.79 s |
| 3 | 0.75 s |
| 4 | 0.80 s |
| 5 | 0.77 s |
| 6 | 0.72 s |
| 7 | 0.81 s |
| 8 | 2.00 s |
| 9 | 0.66 s |
| 10 | 0.68 s |


Mean: 1.386 seconds
Variance: 2.651 seconds squared
Standard Deviation: 1.628 seconds

Lower Quartile (Q1): 0.72 seconds
Median (Q2): 0.78 seconds
Upper Quartile (Q3): 0.81 seconds
Interquartile Range: 0.09 seconds

### 500 Topic Summary Models

| Attempt | Server response wait time |
| ---- | ---- |
| 1 | 3.82 s |
| 2 | 3.50 s |
| 3 | 4.76 s |
| 4 | 3.32 s |
| 5 | 3.35 s |
| 6 | 4.55 s |
| 7 | 8.43 s |
| 8 | 3.30 s |
| 9 | 3.18 s |
| 10 | 8.22 s |


Mean: 4.643 seconds
Variance: 4.057 seconds squared
Standard Deviation: 2.014 seconds

Lower Quartile (Q1): 3.32 seconds
Median (Q2): 3.66 seconds
Upper Quartile (Q2): 4.76 seconds
Interquartile Range: 1.44 seconds

### 1000 Topic Summary Models

| Attempt | Server response wait time |
| ---- | ---- |
| 1 | 9.24 s |
| 2 | 7.55 s |
| 3 | 12.32 s |
| 4 | 7.52 s |
| 5 | 7.35 s |
| 6 | 12.30 s |
| 7 | 7.28 s |
| 8 | 12.37 s |
| 9 | 7.35 s |
| 10 | 7.46 s |


Mean: 9.074 seconds
Variance: 5.371 seconds squared
Standard Deviation: 2.318 seconds

Lower Quartile (Q1): 7.35 seconds
Median (Q2): 7.535 seconds
Upper Quartile (Q3): 12.3 seconds
Interquartile Range: 4.95 seconds

## How many models for the handler to take ~30 seconds?

Let $t$ be the number of seconds that the handler should take, so $t=30\text{ s}$. There are two appropriate measures of central tendency that we could use to provide different answers to the above question: mean and median.

### Mean of each test

The arithmetic mean takes each score into account, including scores with extreme values. 

$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{mean time}} = 30\text{ s} \cdot \frac{100\text{ models}}{1.386\text{ s}} = 2165\text{ models}$
$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{mean time}} = 30\text{ s} \cdot \frac{500\text{ models}}{4.643\text{ s}} = 3231\text{ models}$
$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{mean time}} = 30\text{ s} \cdot \frac{1000\text{ models}}{9.074\text{ s}} = 3306\text{ models}$

It could take anywhere from 2165 to 3306 models for the handler to take 30 seconds. There's a drastic difference between the results in the 100 models test and 500 models test compared to the difference between the results in the 500 models test and 1000 models test. I'm not sure what could be the cause of this. 

The arithmetic mean of the attempted calculations is 2901 models, while the median is 3231 models. 

### Median of each test

The median does not take every single score into account, but it also means that the median isn't influenced by scores with extreme values.

$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{median time}} = 30\text{ s} \cdot \frac{100\text{ models}}{0.78\text{ s}} = 3846\text{ models}$
$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{median time}} = 30\text{ s} \cdot \frac{500\text{ models}}{3.66\text{ s}} = 4098\text{ models}$
$\text{number of models} = t \cdot \frac{\text{number of saved models}}{\text{median time}} = 30\text{ s} \cdot \frac{1000\text{ models}}{7.535\text{ s}} = 3981\text{ models}$

It could take anywhere from 3846 models to 4098 models for the handler to take 30 seconds. The arithmetic mean of the attempted calculations is 3975 models, while the median is 3981 models.

### Other Observations

Since the mean for each test is greater than its median, this implies that the scores with extreme values are greater than the scores with a closer-to-average value.

Reply:
OK, that actually seems reasonably safe -- it'll be a long time before we get to 1000 topics :) Thanks for analyzing!